### PR TITLE
Improve OpenTelemetry docs, fix #5688

### LINKED ISSUE #5688: OpenTelemetry documentation is misleading with the deprecated Elastic exporter more visible than the OTLP exporter
As shown on https://discuss.elastic.co/t/servermap-module-data-exceptions/278012/7 , the current [Elastic + OpenTelemetry documentation](https://www.elastic.co/guide/en/apm/get-started/current/open-telemetry-elastic.html) is misleading with the deprecated Elastic exporter more visible than the OTLP exporter.

We should clearly highlight that the OpenTelemetry Collector Exporter for Elastic is deprecated and primarily document the OTLP exporter.

We will fix this just after merging https://github.com/elastic/apm-server/pull/5674.

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
* Limit docs on the deprecated OpenTelemetry Collector exporter for Elastic in this guide and focus on the new OTLP solution
* Introduce a "Known limitations" section

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
@EamonnTP @bmorelli25 could you please explain me how we will port this documentation on the 7.14 and later branches?

**Comment by axw** [comment]
@cyrille-leclerc can you please add a known limitation about extrapolating metrics when sampling? i.e. if your OTel traces are sampled < 100%, throughput metrics in the UI will be off. This is because there's no record of the sampling rate in span data; we would need https://github.com/open-telemetry/oteps/pull/148 for this.

**Comment by cyrille-leclerc** [comment]
Thanks @axw , do you mind creating an "Enhancement" request to track the limitation related to metrics accuracy when sampling? maybe in the elastic/apm repo?
Please add the OpenTelemetry label

**Comment by cyrille-leclerc** [comment]
@axw I created it see https://github.com/elastic/apm/issues/472

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 1]
```diff
@@ -20,36 +20,37 @@ Elastic OpenTelemetry integrations allow you to reuse your existing OpenTelemetr
...
-* <<open-telemetry-elastic-exporter,Elastic exporter on the OpenTelemetry collector>>
-
 [[open-telemetry-elastic-protocol]]
 ==== APM Server native support of OpenTelemetry protocol
 
+
+IMPORTANT: The https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#legacy-opentelemetry-collector-exporter-for-elastic[OpenTelemetry Collector exporter for Elastic]
+is deprecated since Elastic observability 7.13 and replaced by the native support of the OpenTelemetry Line Protocol in
+Elastic Observability (aka OTLP). See the
+https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#migration[migration guide].
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:30]:
```suggestion
IMPORTANT: The https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#legacy-opentelemetry-collector-exporter-for-elastic[OpenTelemetry Collector exporter for Elastic]
was deprecated in 7.13 and replaced by the native support of the OpenTelemetry Line Protocol in
Elastic Observability (OTLP). To learn more, see 
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#migration[migration].
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 2]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
-Please note the required space between `Bearer` and `apm_secret_token`, and `APIKey` and `api_key`.
+Please note the required space between `Bearer` and `an_apm_secret_token`, and `APIKey` and `an_api_key`.
 
 | `OTEL_EXPORTER_OTLP_CERTIFICATE` | Certificate for TLS credentials of the gRPC client. (optional)
 
 |===
 
-You are now ready to collect <<open-telemetry-elastic-traces-metrics,traces and metrics>>, <<open-telemetry-elastic-verify,verify metrics>>,
+You are now ready to collect traces and <<open-telemetry-elastic-metrics,metrics>>, <<open-telemetry-elastic-verify,verify metrics>>,
 and <<open-telemetry-elastic-kibana,visualize metrics>> in {kib}.
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:76]:
```suggestion
You are now ready to collect traces and <<open-telemetry-elastic-metrics,metrics>> before <<open-telemetry-elastic-verify,verifying metrics>>
and <<open-telemetry-elastic-kibana,visualizing metrics>> in {kib}.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 3]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
-They can be deployed as often as necessary and scaled up or out. Deployment planning resources are available in
-OpenTelemetry's {ot-collector}[Getting Started] documentation and {ot-scaling}[Collector Performance] research.
-
-You can download the latest release of the collector from the {ot-contrib}/releases[GitHub releases page].
-The Elastic exporter lives in the {ot-contrib}[`opentelemetry-collector-contrib` repository].
+[[open-telemetry-collector]]
+===== Connect OpenTelemetry Collector instances
 
-Docker images are available on {ot-dockerhub}[dockerhub]:
+When using the OpenTelemetry Collector instances in your architecture, you can connect them to Elastic Observability using the OTLP exporter.
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:81]:
```suggestion
Using the OpenTelemetry collector instances in your architecture, you can connect them to Elastic Observability using the OTLP exporter.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 4]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+  pipelines:
+    traces:
+      receivers: [otlp]
+      exporters: [logging, otlp/elastic]
+    metrics:
+      receivers: [otlp]
+      exporters: [logging, otlp/elastic]
 ----
+<1> The receivers that matter to your use case such as
+the https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver[OTLP receiver] to forward data emitted by APM agents or the https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/hostmetricsreceiver[Host Metrics receiver],
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:117]:
```suggestion
<1> The receivers, such as
the https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver[OTLP receiver], that forward data emitted by APM agents or the https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/hostmetricsreceiver[host metrics receiver].
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 5]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+<2> The processors that matter to your use case ; Elastic recommends using the https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md[Batch processor] and suggests also using the https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiter/README.md[Memory Limiter processor]. More recommandations are available on https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/README.md#recommended-processors[OpenTelemetry Collector / Recommended Processors],
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:118]:
```suggestion
<2> We recommend using the https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md[Batch processor] and also suggest using the https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiter/README.md[memory limiter processor]. For more information, see https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/README.md#recommended-processors[Recommended processors].
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 6]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+<3> The https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/loggingexporter[Logging exporter] is convenient for troubleshooting, supported logging levels: `debug`, `info`, `warn`, and `error`,
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:119]:
```suggestion
<3> The https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/loggingexporter[logging exporter] is helpful for troubleshooting and supports various logging levels: `debug`, `info`, `warn`, and `error`.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 7]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+<4> Elastic Observability endpoint configuration. See https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter[OpenTelemetry Collector > OTLP gRPC exporter] documentation for more details.
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:120]:
```suggestion
<4> Elastic Observability endpoint configuration. To learn more, see https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter[OpenTelemetry Collector > OTLP gRPC exporter].
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 8]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+<5> Hostname and port of the APM Server endpoint (e.g. `elastic-apm-server:8200`),
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:121]:
```suggestion
<5> Hostname and port of the APM Server endpoint. For example, `elastic-apm-server:8200`.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 9]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+<6> Credential for Elastic APM {apm-server-ref-v}/secret-token.html[secret token authorization] (e.g. `Authorization: "Bearer a_secret_token"`) or {apm-server-ref-v}/api-key.html[API key authorization] (e.g. `Authorization: "ApiKey an_api_key"`),
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:122]:
```suggestion
<6> Credential for Elastic APM {apm-server-ref-v}/secret-token.html[secret token authorization] (`Authorization: "Bearer a_secret_token"`) or {apm-server-ref-v}/api-key.html[API key authorization] (`Authorization: "ApiKey an_api_key"`).
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 10]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
+<7> Note that environment-specific configuration parameters can be conveniently passed in as environment variables as documented https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables[here] (e.g. `ELASTIC_APM_SERVER_ENDPOINT` and `ELASTIC_APM_SERVER_TOKEN`).
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:123]:
```suggestion
<7> Environment-specific configuration parameters can be conveniently passed in as environment variables documented https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables[here] (e.g. `ELASTIC_APM_SERVER_ENDPOINT` and `ELASTIC_APM_SERVER_TOKEN`).
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 11]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
 * Sample Terraform code: https://github.com/cyrille-leclerc/my-serverless-shopping-cart/tree/main/checkout-function/deploy
 * Note that the Terraform code to manage the HTTP API Gateway (https://github.com/cyrille-leclerc/my-serverless-shopping-cart/tree/main/utils/terraform/api-gateway-proxy[here]) is copied from the official OpenTelemetry Lambda sample https://github.com/open-telemetry/opentelemetry-lambda/tree/e72467a085a2a6e57af133032f85ac5b8bbbb8d1/utils[here]
 
+[[elastic-open-telemetry-known-limitations]]
+==== Known limitations
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:311]:
```suggestion
==== Limitations in 7.13
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 12]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+
+===== Known limitations in Elastic Observability 7.13
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:313]: This section title can be removed.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 13]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+
+* OpenTelemetry traces
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:315]:
```suggestion
[[elastic-open-telemetry-traces-limitations]]
===== OpenTelemetry traces

```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 14]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** Traces of applications using `messaging` semantics might be wrongly displayed or not shown at all in Elastic APM UI. You might see only `spans` coming from such services, but no `transaction`.
+*** See issue https://github.com/elastic/apm-server/issues/5094[[OpenTelemetry\] Map OpenTelemetry `messaging.*` semantic conventions to Elastic Common Schema #5094]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:317]:
```suggestion
* Traces of applications using `messaging` semantics might be wrongly displayed or not shown in the APM UI. You may only see `spans` coming from such services, but no `transaction` https://github.com/elastic/apm-server/issues/5094[#5094]
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 15]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** Inability to see Stack traces in spans or in general, arbitrary span events for applications instrumented with OpenTelemetry
+*** See issue https://github.com/elastic/apm-server/issues/4715[[OpenTelemetry\] Support OpenTelemetry Span Events #4715]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:319]:
```suggestion
* Inability to see Stack traces in spans or, in general, arbitrary span events for applications instrumented with OpenTelemetry https://github.com/elastic/apm-server/issues/4715[#4715]
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 16]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** Inability in APM views to view the "Time Spent by Span Type"
+** See issue https://github.com/elastic/apm-server/issues/5747[[OpenTelemetry\] The graph "Time spent by span type" is empty #5747]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:321]:
```suggestion
* Inability in APM views to view the "Time Spent by Span Type" https://github.com/elastic/apm-server/issues/5747[#5747]
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 17]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** Metrics derived from traces (Throughput, Latency and Errors) are not accurate when traces are sampled before being ingested by Elastic Observability (ie by an OpenTelemetry Collector or OpenTelemetry APM agent or SDK)
+*** See issue https://github.com/elastic/apm/issues/472[[OpenTelemetry\] Metrics derived from traces (Throughput, Latency and Errors) are not accurate when traces are sampled before being ingested by Elastic Observability #472]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:323]:
```suggestion
* Metrics derived from traces (throughput, latency, and errors) are not accurate when traces are sampled before being ingested by Elastic Observability (ie by an OpenTelemetry Collector or OpenTelemetry APM agent or SDK) https://github.com/elastic/apm/issues/472[#472]

```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 18]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+* OpenTelemetry Metrics
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:324]:
```suggestion
[[elastic-open-telemetry-metrics-limitations]]
===== OpenTelemetry metrics

```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 19]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** OpenTelemetry histogram metrics, https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/DoubleValueRecorder.html[`DoubleValueRecorder`]
+and https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/LongValueObserver.html[`LongValueRecorder`], are not yet supported
+*** See issue https://github.com/elastic/apm-server/issues/3195[Histogram Support #3195]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:327]:
```suggestion
* OpenTelemetry histogram metrics, https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/DoubleValueRecorder.html[`DoubleValueRecorder`]
and https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/LongValueObserver.html[`LongValueRecorder`], are not yet supported https://github.com/elastic/apm-server/issues/3195[#3195]
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 20]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** Inability to see JVM metrics in Elastic APM UI for Java applications instrumented with OpenTelemetry
+*** See issue https://github.com/elastic/apm-server/issues/4919[Visualize JVM health indicators collected by the OpenTelemetry Java Agent #4919]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:329]:
```suggestion
* Inability to see JVM metrics in Elastic APM UI for Java applications instrumented with OpenTelemetry https://github.com/elastic/apm-server/issues/4919[#4919]
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 21]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** Inability to see Host metrics in Elastic Metrics Infrastructure view when using the OpenTelemetry Collector Host Metrics receiver
+*** See issue https://github.com/elastic/apm-server/issues/5310[Map the metrics of the OpenTelemetry Collector Host Metrics Receiver to ECS and store them in the metrics- indices #5310]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:331]:
```suggestion
* Inability to see host metrics in Elastic Metrics Infrastructure view when using the OpenTelemetry Collector host metrics receiver https://github.com/elastic/apm-server/issues/5310[#5310]

```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 22]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+* OpenTelemetry logs
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:332]:
```suggestion
[[elastic-open-telemetry-logs-limitations]]
===== OpenTelemetry logs

```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 23]
```diff
@@ -402,5 +307,31 @@ Here is an example of AWS Lambda Java function managed with Terraform and the ht
...
+** OpenTelemetry logs are not yet supported
+*** See issue https://github.com/elastic/apm-server/issues/5491[[OpenTelemetry\] add support for receiving logs #5491]
```
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:334]:
```suggestion
* OpenTelemetry logs are not yet supported https://github.com/elastic/apm-server/issues/5491[#5491]
```

**Review by EamonnTP — COMMENTED** [review]
Looks great, just some minor suggestions. Note that "Positive Rate" will need updated to "Counter Rate" in the Visualize in Kibana section.

**Comment by EamonnTP** [comment]
> @EamonnTP @bmorelli25 could you please explain me how we will port this documentation on the 7.14 and later branches?

@cyrille-leclerc We would use the back port tool for back porting PRs. This is something Brandon and I could help out with. After merging, I take it this PR needs back ported to the `master`, `7.x`, and `7.14` branches?

**Comment by EamonnTP** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 12 above)
Comment (bmorelli25) [reply docs/guide/opentelemetry-elastic.asciidoc:313]:
```suggestion
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 24]
```diff
@@ -20,36 +20,37 @@ Elastic OpenTelemetry integrations allow you to reuse your existing OpenTelemetr
...
+Elastic Observability (OTLP). To learn more, see 
+https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#migration[migration].
+
 Elastic APM Server natively supports the OpenTelemetry protocol.
 This means trace data and metrics collected from your applications and infrastructure can
-be sent directly to APM Server using the OpenTelemetry protocol.
+be sent directly to Elastic APM Server using the OpenTelemetry protocol.
 
-image::images/open-telemetry-protocol-arch.png[OpenTelemetry Elastic protocol architecture diagram]
+image::images/open-telemetry-protocol-arch.png[OpenTelemetry Elastic architecture diagram]
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:36]: Great diagram

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 25]
```diff
@@ -61,173 +62,77 @@ java -javaagent:/path/to/opentelemetry-javaagent-all.jar \
...
-        exporters:
-          - elastic <3>
-      traces:
-        receivers:
-          - otlp
-        processors:
-          - batch
-        exporters:
-          - elastic <4>
++
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:83]:
```suggestion
```

**Review by bmorelli25 — APPROVED** [review]
I committed the rest of @EamonnTP's suggestions as it looks like you just missed the expand comment button. Otherwise, this looks good! Thanks, Cyrille.

**Comment by bmorelli25** [comment]
I'll merge and cherry-pick when the tests are green.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 24 above)
Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:36]: thanks

**Comment by cyrille-leclerc** [comment]
Thanks Brandon

**Pull request merged**